web/src/main/java/org/springframework/security/web/FilterChainProxy.java

```diff
@@ -176,6 +182,8 @@ public void doFilter(ServletRequest request, ServletResponse response,
 			try {
 				request.setAttribute(FILTER_APPLIED, Boolean.TRUE);
 				doFilterInternal(request, response, chain);
+			} catch (RequestRejectedException e) {
```

Comment:
Does this need to be handled in the `else` block (line 194) as well?

Reply:
I don't think so, as it has already been applied in the filter chain before as indicated by the `FILTER_APPLIED` attribute being present, so it is already in a surrounding try-catch, doing it again could call the handler more than once.